build.sbt

```diff
@@ -117,6 +117,29 @@ lazy val root = (project in file("."))
     )
   )
 
+lazy val scalaVersionDependentSettings = Def.settings(
+  libraryDependencies ++= (if (scalaVersion.value.startsWith("2."))
+                             COMPILE(
+                               scalaVersion("org.scala-lang" % "scala-reflect" % _).value
+                             )
+                           else Seq.empty),
+  scalacOptions ++= (if (scalaVersion.value.startsWith("2."))
+                       Seq(
+                         "-Ywarn-macros:after",
+
+                         // Ensure we publish an artifact linked to the appropriate Java std library.
+                         "-release:17",
+
+                         // Implicit resolution debug flags.
+                         "-Vimplicits",
+                         "-Vtype-diffs",
+                         // Make all warnings verbose.
+                         "-Wconf:any:warning-verbose"
```

Comment:
I am not very familiar with **Scala 3**, but are you sure those do not exist there?

Reply:
It seems they don't and their alternatives do not too.

https://github.com/scala/docs.scala-lang/blob/main/_overviews/scala3-migration/options-lookup.md